### PR TITLE
feat: use cobra for the CLI

### PR DESCRIPTION
**Pull request by eddycharly**
Use cobra for the CLI.

### CODE REVIEW

**Comment by eddycharly** [comment]
@springcomp time to build a real cli :)

**Review by springcomp — APPROVED** [review]
I will trust you on this one.

I have one piece of feedback though:

My understanding was that the CLI in this repository was intended as quick and easy CLI tool for test purposes. But it does not necessarily have all the bells and whistles as a proper CLI could have.

In contrast, the CLI in [that repository](https://github.com/jmespath-community/jp) is a full-featured project, that supports console colors, streaming input, etc.

I can onboard you as admin to that project as well if you want.
Or we can merge the two repositories and fold the CLI in this single repository here if that’s better.

What do you think ?

TL;DR; my view is that we should improve the CLI in the other repository instead and keep this one purposefuly simple and refrain from adding features.

**Comment by eddycharly** [comment]
I don't mind collaborating on the other repo :)

IMHO commands should be split, like `jp query`, `jp parse`, etc...
Anyway, switching to cobra is definitely going to help, let's merge this one and see how we can make tools converge.

**Comment by eddycharly** [comment]
`jp` is using `urfave/cli` it's a good starting point.

**Comment by springcomp** [comment]
> I don't mind collaborating on the other repo :)
> 

You are now admin to the `jp` repository as well.
Thanks for your contributions 🙏
